### PR TITLE
Require less data in transaction builders in Auction House

### PR DESCRIPTION
**Pull request by zaxozhu**

### CODE REVIEW

File: packages/js/src/plugins/auctionHouseModule/operations/cancelBid.ts [hunk 1]
```diff
@@ -105,28 +118,42 @@ export const cancelBidBuilder = (
...
-
   // Accounts.
   const tokenAccount = isPublic
     ? findAssociatedTokenAccountPda(
         asset.mint.address,
-        toPublicKey(bid.buyerAddress)
+        toPublicKey(buyerAddress)
       )
-    : bid.asset.token.address;
+    : (asset as SftWithToken | NftWithToken).token.address;
```
Comment (lorisleiva) [inline packages/js/src/plugins/auctionHouseModule/operations/cancelBid.ts:148]: Why do we need the type hint here?

**Review by lorisleiva — CHANGES_REQUESTED** [review]
Just a little comment but otherwise we're looking good. 🙂

File: packages/js/src/plugins/auctionHouseModule/operations/cancelBid.ts (hunk 1 above)
Comment (zaxozhu) [reply packages/js/src/plugins/auctionHouseModule/operations/cancelBid.ts:148]: TypeScript doesn't see typing that comes from isPublic, so It doesn't treat assets as SftWithToken | NftWithToken by default

Comment (lorisleiva) [reply packages/js/src/plugins/auctionHouseModule/operations/cancelBid.ts:148]: Thanks!

**Review by lorisleiva — APPROVED** [review]

**Pull request merged**